internal/provider/service_resource.go

```diff
@@ -577,6 +591,16 @@ func (r *ServiceResource) Update(ctx context.Context, req resource.UpdateRequest
 		resp.Diagnostics.AddError(ErrCreateTimeout, fmt.Sprintf("error occurred while waiting for service reconfiguration, got error: %s", err))
 		return
 	}
+
+	// Update Password if it has changed
+	if !plan.Password.Equal(state.Password) && !plan.Password.IsNull() {
```

Comment:
Possibly a naive question, how do we read the state password (assuming the state is the current state of the terraform service)? We don't provide any endpoints in our stack for retrieving the tsdbadmin password, as we don't store it. Is there a way for the user to modify the state in a separate step from the "plan"?

Reply:
It's just stored in the state in plain text if service managed to get created and we obtained the initial password in service creation call.

Reply:
Gotcha, thanks for clarifying!